### DISCUSSION TITLE
iOS | Jenkins | Xcode couldn't find any provisioning profiles matching

### DISCUSSION THREAD

**Discussion by pushkrajlanjekar** [fastlane/fastlane#18266] [Q&A] [answered]
I am frequently getting this error, tried all possible solutions

❌  error: No profile for team 'MYTEAMID' matching 'Find My Buddy App Store Distribution 20 Feb' found: Xcode couldn't find any provisioning profiles matching 'MYTEAMID/MY_PROVISIONING_PROFILE'. Install the profile (by dragging and dropping it onto Xcode's dock item) or select a different one in the Signing & Capabilities tab of the target editor. (in target 'FindMyBuddy' from project 'FindMyBuddy')[0m
[00:30:54]: ▸ [35m** ARCHIVE FAILED **[0m

Let me know how can I resolve this?

Some Imp notes: When I run fastlane on local, Archive is getting created. Its only giving issue with Jenkins.

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.7                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.24.3 (Apple Git-128)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.7 (19H2)                                        |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.7                                                           |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile.swift`</summary>

```ruby
// This file contains the fastlane.tools configuration
// You can find the documentation at https://docs.fastlane.tools
//
// For a list of all available actions, check out
//
//     https://docs.fastlane.tools/actions
//

import Foundation

class Fastfile: LaneFile {
	func releasefindmybuddyLane() {
        xcversion(version: "11.7")
        desc("Find My Buddy Beta App Release")
		buildApp(workspace: "FindMyBuddy.xcworkspace", scheme: "FindMyBuddy")
		uploadToTestflight(username: "my_apple_id@apple.com")
	}
}

```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.175.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                           | Version      |
| ----------------------------- | ------------ |
| did_you_mean                  | 1.3.0        |
| slack-notifier                | 2.3.2        |
| rouge                         | 2.0.7        |
| xcpretty                      | 0.3.0        |
| terminal-notifier             | 2.0.0        |
| unicode-display_width         | 1.7.0        |
| terminal-table                | 1.8.0        |
| addressable                   | 2.7.0        |
| multipart-post                | 2.0.0        |
| word_wrap                     | 1.0.0        |
| tty-cursor                    | 0.7.1        |
| tty-spinner                   | 0.9.3        |
| artifactory                   | 3.0.15       |
| colored                       | 1.2          |
| highline                      | 1.7.10       |
| commander-fastlane            | 4.4.6        |
| faraday_middleware            | 1.0.0        |
| gh_inspector                  | 1.1.3        |
| rubyzip                       | 2.3.0        |
| security                      | 0.1.3        |
| simctl                        | 1.6.8        |
| uber                          | 0.1.0        |
| declarative-option            | 0.1.0        |
| representable                 | 3.0.4        |
| retriable                     | 3.1.2        |
| mini_mime                     | 1.0.2        |
| httpclient                    | 2.8.3        |
| google-api-client             | 0.38.0       |
| nanaimo                       | 0.3.0        |
| colored2                      | 3.1.2        |
| claide                        | 1.0.3        |
| CFPropertyList                | 3.0.3        |
| atomos                        | 0.1.3        |
| xcodeproj                     | 1.19.0       |
| plist                         | 3.6.0        |
| public_suffix                 | 4.0.6        |
| tty-screen                    | 0.8.1        |
| babosa                        | 1.0.4        |
| excon                         | 0.79.0       |
| unf_ext                       | 0.0.7.7      |
| unf                           | 0.1.4        |
| domain_name                   | 0.5.20190701 |
| http-cookie                   | 1.0.3        |
| ruby2_keywords                | 0.0.4        |
| faraday-net_http              | 1.0.1        |
| faraday                       | 1.3.0        |
| faraday-cookie_jar            | 0.0.7        |
| fastimage                     | 2.2.2        |
| json                          | 2.5.1        |
| mini_magick                   | 4.11.0       |
| naturally                     | 2.2.1        |
| xcpretty-travis-formatter     | 1.0.1        |
| dotenv                        | 2.7.6        |
| bundler                       | 2.2.11       |
| jwt                           | 2.2.2        |
| declarative                   | 0.0.20       |
| multi_json                    | 1.15.0       |
| signet                        | 0.14.1       |
| os                            | 1.1.1        |
| memoist                       | 0.16.2       |
| googleauth                    | 0.15.1       |
| rake                          | 13.0.3       |
| digest-crc                    | 0.6.3        |
| webrick                       | 1.7.0        |
| rexml                         | 3.2.4        |
| google-apis-core              | 0.2.1        |
| google-apis-storage_v1        | 0.2.0        |
| google-apis-iamcredentials_v1 | 0.1.0        |
| google-cloud-errors           | 1.0.1        |
| google-cloud-env              | 1.4.0        |
| google-cloud-core             | 1.5.0        |
| google-cloud-storage          | 1.30.0       |
| emoji_regex                   | 3.2.2        |
| aws-eventstream               | 1.1.0        |
| aws-sigv4                     | 1.2.2        |
| aws-partitions                | 1.428.0      |
| jmespath                      | 1.4.0        |
| aws-sdk-core                  | 3.112.0      |
| aws-sdk-kms                   | 1.42.0       |
| aws-sdk-s3                    | 1.88.1       |
| forwardable                   | 1.2.0        |
| logger                        | 1.3.0        |
| date                          | 2.0.0        |
| stringio                      | 0.0.2        |
| zlib                          | 1.0.0        |
| ipaddr                        | 1.2.2        |
| openssl                       | 2.1.2        |
| ostruct                       | 0.1.0        |
| strscan                       | 1.0.0        |
| fileutils                     | 1.1.0        |
| etc                           | 1.0.1        |
| io-console                    | 0.4.7        |
| libxml-ruby                   | 3.1.0        |
| psych                         | 3.1.0        |
| mutex_m                       | 0.1.0        |
| ox                            | 2.13.2       |
</details>

*generated on:* **2021-02-24**
</details>

**Comment by lucgrabowski** [comment, marked as answer]
Hi, to be able to build your app, you need to install certificate and provisioning profile on the server with jenkins first. It perfectly works for you locally, because you installed both certificate and profile.
You have 3 options
- install manually
- use `match` tool, 
or 
- use `sigh` with `cert`.

A good starting point is docs about code signing: https://docs.fastlane.tools/codesigning/getting-started/

**Reply by BeatrizSantos-Costa** [reply]
Hi, I'm having this problem even using match, can you tell me how I can solve it?

`[35mnote: Build preparation complete[0m
12:24:54  [15:24:54]: ▸ [35mnote: Building targets in dependency order[0m
12:24:54  [15:24:54]: ▸ [35merror: No profiles for 'XXX' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'XXXX'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild. (in target 'XXX' from project 'XXX')[0m
12:24:54  [15:24:54]: 
12:24:54  [15:24:54]: [31m⬆️  Check out the few lines of raw `xcodebuild` output above for potential hints on how to solve this error[0m
12:24:54  [15:24:54]: [33m📋  For the complete and more detailed error log, check the full log at:[0m
12:24:54  [15:24:54]: [33m📋  /Users/XXX/Library/Logs/gym/XXX-XXXX.log[0m`

**Reply by devqueue** [reply]
Hi,  @lucgrabowski would you mind answering this https://github.com/fastlane/fastlane/discussions/21967 please. I am using match but still facing the same issue

**Comment by pushkrajlanjekar** [comment]
Yes, even I am successfully able to run it. Closing the issue.

On Thu, Jun 24, 2021 at 11:42 AM Madaline24 ***@***.***>
wrote:

> I am now able to successfully build. Not sure exactly which step "fixed"
> things, but this was the sequence:
> Tried automatic signing again. No go, so reverted to manual.
> After reverting, I had no Eligible Profiles, all were ineligible. Strange.
> I created a new certificate and profile, imported both. This too was
> "ineligible".
> Removed the iOS platform and re-added it. I had tried this previously
> without luck.
> After doing this, Xcode on its own defaulted to automatic signing. And
> this worked! Success!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/fastlane/fastlane/discussions/18266#discussioncomment-913293>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZMEMFYT3KTMCR3SP2HUPLTULEEPANCNFSM4YDBO2DQ>
> .
>


-- 
With Regards,

Pushkraj P. Lanjekar
